config.c

```diff
@@ -691,6 +691,15 @@ int load_config_file(struct mako_config *config, char *config_arg) {
 		}
 
 		if (line[0] == '[' && line[strlen(line) - 1] == ']') {
+			// Since we hit the end of the previous criteria section, validate
+			// that it doesn't break any rules before moving on.
+			if (criteria != NULL && !validate_criteria(criteria)) {
```

Comment:
This doesn't validate the last criteria section.

Maybe we can just loop over all criteria sections after parsing them?

Reply:
Ah, this is fixed in a subsequent commit. Would be nice to squash these two.

Reply:
Good idea, will do.